src/ui/public/notify/notifier.js

```diff
@@ -93,6 +93,12 @@ function add(notif, cb) {
     notif.actions.forEach(function (action) {
       notif[action] = closeNotif(notif, cb, action);
     });
+  } else if (notif.customActions) {
+    // wrap all of the custom functions in a close
+    notif.customActions = notif.customActions.map(action => {
+      action.callback = closeNotif(notif, action.callback, action.key);
```

Comment:
Why use `key` in the definition? From the outside, that parameter is really just defining the button text, isn't it? Couldn't it be `text`, like:

```js
customActions: [{ text: 'Do it!', callback: doTheThing }]
```